wordpress-mobile/WordPress-iOS
# Zendesk Mobile: show Zendesk support options instead of Helpshift

Fixes #9143 

**NOTES:**
- Your local `.wpcom_app_credentials` must be up-to-date as it contains Zendesk credentials. If you have not, refer to the Field Guide for instructions.
- All of the following views will show Zendesk. However, if you're using a jetpack or self-hosted account, the support options won't actually do anything yet.

**To test:**
Below is a list of locations that will show the Zendesk Support options instead of Helpshift. Because most depend on 'no results' scenarios, I've included hacks to trigger them. It's not pretty, but it works.

- Me > Help & Support
- Blog details > Activity > No results > Contact Support button
  - hack: `ActivityListViewModel` - always return the "Oops" error view for `var noResultsViewModel`.
- Blog details > Settings > Time Zone > No results > Contact Support button
  - hack: `TimeZoneSelectorViewController` - always return the "Oops" error view for `var noResultsViewModel`.
- Blog details > Plans > no results > Contact Support button
  - hack: `PlanListViewModel` - always return the "Oops" error view for `var noResultsViewModel`.
- Blog details > Plans > select a plan > no results > Contact Support button
  - hack: `PlanDetailViewModel` - always return the "Oops" error view for `var noResultsViewModel`.
- `WPError:showAlert` methods > Need Help button
  - hack: disable network connection. Launch app. Should get 'Unable to Sync' error.
  - hack: add an error to `BlogListViewController:viewWillAppear`. Ex: `[WPError showAlertWithTitle:@"error title" message:@"error message"];`
- Blog details > Plugins list > no results > Contact Support button (requires a site with Business plan)
  - hack: `PluginListViewModel` - always return the "Oops" error view for `var noResultsViewModel`.


Distinguishing between Zendesk and Helpshift Support views is pretty simple - Zendesk has an additional 'My Tickets' option.

**Zendesk view:**
<img width="250" alt="zendesk" src="https://user-images.githubusercontent.com/1816888/39149918-e31739b6-46fd-11e8-8c0b-6bfa545ce4ef.png">


**Helpshift view:**
<img width="250" alt="helpshift" src="https://user-images.githubusercontent.com/1816888/39149937-f20925f6-46fd-11e8-9b75-756ae97803a5.png">
